packages/types/index.d.ts

```diff
@@ -23,20 +23,49 @@ export type SnapRpcHandler = (
 export interface SnapProvider extends MetaMaskInpageProvider {
   registerRpcMessageHandler: (handler: SnapRpcHandler) => void;
 }
-type SnapName = string;
+
+export type SnapManifest = {
+  version: string;
+  proposedName: string;
+  description: string;
+  repository: {
+    type: string;
+    url: string;
+  };
+  source: {
+    shasum: string;
+    location: {
+      npm: {
+        filePath: string;
+        packageName: string;
+        registry: string;
+      };
+    };
+  };
+  initialPermissions: Record<string, Record<string, Json>>;
+  manifestVersion: '0.1';
+};
```

Comment:
we could generate this type from the JSON Schema with this tool: https://github.com/json-schema-tools/transpiler